### PR TITLE
Added batch iteration into input and target variables (#285)

### PR DESCRIPTION
**Pull request by keitakurita**
Now, the batch can be iterated into input and target variables like
```
>>> (x1, x2, x3), y = batch
```

where `x1` to `x3` are inputs to a downstream model and `y` is the label data for this batch.

### CODE REVIEW

File: torchtext/data/batch.py [hunk 1]
```diff
@@ -58,6 +62,20 @@ def __str__(self):
     def __len__(self):
         return self.batch_size
 
+    def _get_vars(self, target):
```
Comment (mttk) [inline torchtext/data/batch.py]: Could this be changed to some enumeration and/or commented (ex `target="outputs"`, or just add a docstring)

File: torchtext/data/batch.py [hunk 2]
```diff
@@ -58,6 +62,20 @@ def __str__(self):
...
+        fields = self.target_fields if target else self.input_fields
+        vars = tuple(getattr(self, f) for f in fields)
+        if len(vars) == 0:
```
Comment (mttk) [inline torchtext/data/batch.py]: Also brief inline comments on the cases here

**Comment by keitakurita** [comment]
Hi, thanks for the review! I changed the code to reflect my intent more clearly, I think this should resolve the above problems, but if anything can be improved please tell me :)

**Comment by mttk** [comment]
Awesome, thanks, everything seems good now!

**Comment by SudoSharma** [comment]
Hey everyone, I ran into the same issue as @raheelqader in #422, with the "is_target" attribute being referenced for a "RawField" instance. I'd like to suggest we add an "is_target" attribute to "RawField" and set it to False by default. 

I'd like to contribute, but this is my first time. Should I just go ahead and do a PR once I've made the fix? Thanks! Looking forward to contributing.

**Comment by mttk** [comment]
Hey, sure, feel free to submit a PR!

**Pull request merged**